File: factors.c
#include <stdio.h>

void factorize(int n)
{
    int i;
    for (i = 2; i <= n / 2; i++)
    {
        if (n % i == 0) {
            printf("%d=%d*%d\n", n, i, n / i);
            return;
        }
    }
    printf("%d=%d*%d\n", n, 1, n);
}

int main(int argc, char *argv[]) {
    if (argc != 2) {
        printf("Usage: %s <file>\n", argv[0]);
        return 1;
    }

    FILE *file = fopen(argv[1], "r");
    if (!file) {
        perror("Error opening file");
        return 1;
    }

    int num;
    while (fscanf(file, "%d", &num) == 1) {
        factorize(num);
    }

    fclose(file);
    return 0;
}

File: factors.py
import sys

def factorize(num):
    factors = []
    divisor = 2
    while num > 1:
        while num % divisor == 0:
            factors.append(divisor)
            num //= divisor
        divisor += 1
    return factors

def find_factors(num):
    factors = factorize(num)
    if len(factors) == 1:
        print(f'{num}={factors[0]}*{factors[0]}')
    else:
        print(f'{num}={factors[0]}*{factors[-1]}')

def main(filename):
    with open(filename, 'r') as file:
        for line in file:
            num = int(line.strip())
            find_factors(num)

if __name__ == '__main__':
    if len(sys.argv) != 2:
        print('Usage: factors <file>')
        sys.exit(1)

    filename = sys.argv[1]
    main(filename)
